main/exercise/calculated_answer.class.php

```diff
@@ -188,62 +281,66 @@ function updateBlanks(e) {
-                $auxAnswer = $answer;
-                $auxFormula = $formula;
-                $nb = preg_match_all('/\[[^\]]*\]/', $auxAnswer, $blanks);
-                if ($nb > 0) {
-                    for ($i = 0; $i < $nb; ++$i) {
-                        $blankItem = $blanks[0][$i];
-                        $replace = array("[", "]");
-                        $newBlankItem = str_replace($replace, "", $blankItem);
-                        $newBlankItem = "[".trim($newBlankItem)."]";
-                        // take random float values when one or both edge values have a decimal point
-                        $randomValue =
-                            (strpos($lowestValues[$i], '.') !== false ||
-                            strpos($highestValues[$i], '.') !== false) ?
-                            mt_rand($lowestValues[$i] * 100, $highestValues[$i] * 100) / 100 : mt_rand($lowestValues[$i], $highestValues[$i]);
-                        $auxAnswer = str_replace($blankItem, $randomValue, $auxAnswer);
-                        $auxFormula = str_replace($blankItem, $randomValue, $auxFormula);
-                    }
-                    $math = new EvalMath();
-                    $result = $math->evaluate($auxFormula);
-                    $result = number_format($result, 2, ".", "");
-                    // Remove decimal trailing zeros
-                    $result = rtrim($result, "0");
-                    // If it is an integer (ends in .00) remove the decimal point
-                    if (mb_substr($result, -1) === ".") {
-                        $result = str_replace(".", "", $result);
-                    }
-                    // Attach formula
-                    $auxAnswer .= " [".$result."]@@".$formula;
-                }
-                $this->save();
-                $objAnswer = new Answer($this->id);
-                $objAnswer->createAnswer($auxAnswer, 1, '', $this->weighting, '');
-                $objAnswer->position = array();
-                $objAnswer->save();
+            )
+        );
+
+        $submitValues = $form->getSubmitValues();
+        $rawDataBlank = [];
+        $rawDataFormula = [];
+        foreach ($submitValues as $submitName => $submitValue) {
+            // blank unknown starts with blank
+            // formula unknown starts with formula
+            if (preg_match("/^blank_([^_]+)_(.+)/", $submitName, $matches)) {
+                $rawDataBlank[$matches[2]][$matches[1]] = $submitValue;
+            } else if (preg_match("/^formula_([^_]+)_(.+)/", $submitName, $matches)) {
+                $rawDataFormula[$matches[2]][$matches[1]] = $submitValue;
             }
         }
+
+        // build formula entry for DB
+        // describes formulas and blanks informations
+        $formulaDB = "";
+        $weighting = 0; // global weighting of the question
+        foreach ($rawDataFormula as $formulaName => $formulaInfo) {
+            $formulaDB .=
+                "=".$formulaName.":".
```

Comment:
Single quotes (to avoid confusing the interpreter into looking for something to interpret)